File: GW2-Unity/GW2/Assets/Scripts/UtilityClasses/Edge.cs
using System.Numerics;

namespace DelaunayVoronoi
{
    public class Edge {


        public Point Point1 { get; }
        public Point Point2 { get; }
        public Point Centre1 { get; set; }

        public Edge(Point point1, Point point2, Point centre1) {
            Point1 = point1;
            Point2 = point2;
            Centre1 = centre1;
        }

        public Edge(Point point1, Point point2) {
            Point1 = point1;
            Point2 = point2;
        }

        public void setCenter(Point centre1) {
            Centre1 = centre1;
        }

        public override bool Equals(object obj) {
            if (obj == null) return false;
            if (obj.GetType() != GetType()) return false;
            var edge = obj as Edge;

            var samePoints = Point1 == edge.Point1 && Point2 == edge.Point2;
            var samePointsReversed = Point1 == edge.Point2 && Point2 == edge.Point1;
            return samePoints || samePointsReversed;
        }

        public override int GetHashCode(){
            int hCode = (int)Point1.X ^ (int)Point1.Y ^ (int)Point2.X ^ (int)Point2.Y;
            return hCode.GetHashCode();
        }
    }
}

File: guild-wars-web/server/engine/voronoi/Edge.js
export default class Edge {
  constructor(point1, point2) {
    this.point1 = point1
    this.point2 = point2
    this.centre1 = null
  }

  equals(other) {
    if (!other) return false
    const samePoints = this.point1 === other.point1 && this.point2 === other.point2
    const samePointsReversed = this.point1 === other.point2 && this.point2 === other.point1
    return samePoints || samePointsReversed
  }

  getCanonicalKey() {
    // Create a stable key for this edge (order-independent)
    const id1 = this.point1._id
    const id2 = this.point2._id
    return id1 < id2 ? `${id1}_${id2}` : `${id2}_${id1}`
  }

  toString() {
    return `Edge(${this.point1}, ${this.point2})`
  }
}
